fiducial_slam/nodes/fiducial_slam_node.py

```diff
@@ -408,6 +410,12 @@ def updatePose(self, tfs, imageTime):
                                                 quat, thisvar)
                 variance = v1
 
+        if self.medianFilterSamples > 0:
+            self.positionHistory.append(position)
+            if len(self.positionHistory) > self.medianFilterSamples:
+                self.positionHistory.pop(0)
+            position = numpy.median(self.positionHistory, axis=0)
```

Comment:
It takes the median across one axis of the positionHistory matrix, which gives us a 3D point.  If we don't specify axes it gives us a scalar which is the median of all the elements. 

$ python
Python 2.7.12 (default, Nov 19 2016, 06:48:10)
[GCC 5.4.0 20160609] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import numpy
>>> m = []
>>> m.append([1, 2, 3])
>>> m.append([2, 3, 4])
>>> numpy.median(m, axis=0)
array([ 1.5,  2.5,  3.5])
>>> numpy.median(m)
2.5
>>> numpy.median(m, axis=1)
array([ 2.,  3.])
>>>

https://docs.scipy.org/doc/numpy/reference/generated/numpy.median.html